src/components/ItemDisplay.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useEffect } from 'react';
+import logo from '../assets/one.jpg';
+import { format } from 'date-fns';
+
+const ItemDisplay = ({ setShowModal, closeModal, details }) => {
+	const mainRef = useRef(null);
+
+	useEffect(() => {
+		mainRef.current = document.getElementById('modal');
+
+		window.addEventListener('click', function (e) {
+			if (e.target === mainRef.current) {
+				setShowModal(false);
+			}
+		});
+	}, [setShowModal]);
```

Comment:
There are better ways to do this. Do you know that react gives a [`ref` attribute](https://react.dev/learn/manipulating-the-dom-with-refs) to DOM elements? Check that out. 

But you wouldn't need it because you only need to handle the `onClick` for the div, and you can do that directly without using refs.

Reply:
Yes I am aware of the ref attribute. I didn't need it here to reference the `main` element. I also just realized that since I am not using the `ref`, I don't need `useRef`

This has now been updated by accessing the element id property and saving it in a variable